presidio-analyzer/presidio_analyzer/nlp_engine/nlp_engine_provider.py

```diff
@@ -84,11 +84,20 @@ def create_engine(self) -> NlpEngine:
             )
         try:
             nlp_engine_class = self.nlp_engines[nlp_engine_name]
-            nlp_engine_opts = {
-                m["lang_code"]: m["model_name"]
-                for m in self.nlp_configuration["models"]
-            }
-            engine = nlp_engine_class(nlp_engine_opts)
+            nlp_models = self.nlp_configuration["models"]
+
+            ner_model_configuration = self.nlp_configuration.get(
+                "ner_model_configuration"
+            )
+            if ner_model_configuration:
+                ner_model_configuration = NerModelConfiguration.from_dict(
+                    ner_model_configuration
+                )
+
+            engine = nlp_engine_class(
+                models=nlp_models, ner_model_configuration=ner_model_configuration
+            )
+            engine.load()
```

Comment:
Would it make sense to include a call to load inside init such that the user does not need to specify engine load at any point?

Reply:
This separation is useful if a user wants to create an NlpEngine in a different way, or when they wish to inject a model object instead of always loading it as part of instantiation.  It also reduces the instantiation time (helpful in tests and mocks) at the expense of taking more time on the first run. 

Do you think it should be automatically loaded? I don't have any strong preferences here.

Reply:
Got it, thank you for describing the scenarios where this is helpful. Let's keep it as-is then.